test/e2e/connectivity_test.go

```diff
@@ -185,10 +187,67 @@ func TestPodConnectivityAfterAntreaRestart(t *testing.T) {
 	data.runPingMesh(t, podNames)
 }
 
-// TestOVSRestart checks that when OVS restarts unexpectedly the Antrea agent takes care of
-// replaying flows. More precisely this tests check that Pod connectivity is not broken after a
-// restart.
+// TestOVSRestart verifies that the datapath is not impacted when the Antrea Agent Pod is restarted
+// (e.g. as part of a RollingUpdate). The test sends ICMP messages every 0.1s and checks that there
+// is no packet loss during the restart. This test does not apply to the userspace ndetdev datapath,
+// since in this case the datapath functionality is implemented by the ovs-vswitchd daemon itself.
 func TestOVSRestart(t *testing.T) {
+	skipIfProviderIs(t, "kind", "test not valid for the netdev datapath type")
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	workerNode := workerNodeName(1)
+	t.Logf("Creating two busybox test Pods on '%s'", workerNode)
+	podNames, podIPs, cleanupFn := createTestBusyboxPods(t, data, 2, workerNode)
```

Comment:
From my understanding, after ovs-vswitchd starts, the userspace flows and dp flows should be synced in some way so that the cached dp flows will be flushed, I suspect the packet loss is 0 is because the default normal flow can forward the packet too as they are in same VLAN. If I'm correct, that could mean if the Pods are on different Nodes, the packet loss won't be 0. Have you already tried different Nodes? If not, I can take a try.

Reply:
Yes you are correct. There is something else at play which I haven't investigated yet: when the OVS daemons are stopped, the VXLAN device (vxlan_sys_4789) goes away, which is something else that needs to be investigated, as it breaks inter-Node connectivity even before the daemons start again. I suggest merging this PR since it is a step in the right direction, but I will also open an issue to keep track of this.

Reply:
BTW, I have updated the commit message and added some comments to the test code to clarify this.

Reply:
@antoninbas Thanks for the clarification!